src/zircon/apis/chunkserver.go

```diff
@@ -0,0 +1,76 @@
+package apis
+
+import "time"
+
+// The version number of a chunk
+type Version uint64
+// An offset within a chunk
+type Offset uint32
+// A length of data within a chunk
+type Length uint32
+// A chunk identifier, not directly exposed to normal clients
+type ChunkNum uint64
+
+// 8 MiB, the maximum size of a chunk stored on the chunkserver
+const MaxChunkSize Length = 8 * 1024 * 1024
+
+// Represents "any version is valid" when passed as a chunk version number
+const AnyVersion Version = 0
+
+// An opaque reference to a particular chunk.
+// Does not include which server to look on, but does include the chunk ID.
+// This acts as a bearer token -- whoever has a copy of it has permission to perform operations on this chunk.
+type ChunkReference struct {
+	Chunk       ChunkNum
+	IsReadWrite bool
+	// TODO: figure this security part out
```

Comment:
Probably remove this bit for now and add it later with more thought towards security